### ISSUE TITLE
cra-append-sw not working

### ISSUE THREAD

**Issue by hungdt-ibl** [bbhlondon/cra-append-sw#12]
i added : 
```
"start": "npm-run-all -p watch-css start-js && cra-append-sw --mode dev ./firebase-messaging-sw.js",
 "build": "npm run build-css && react-scripts build && ra-append-sw ./firebase-messaging-sw.js",
```
with ```./firebase-messaging-sw.js``` place in ```/public/firebase-messaging-sw.js```
but no service worker registed after : npm run start

**Comment by 8ctopotamus** [comment]
You may not need to use cra-append-sw. I was able to register the addition service worker by placing it in my public directory as you have done. 

Then in my src/index.js:
```
if ('serviceWorker' in navigator) {
  navigator.serviceWorker.register('../firebase-messaging-sw.js')
  .then(function(registration) {
    console.log('Registration successful, scope is:', registration.scope);
  }).catch(function(err) {
    console.log('Service worker registration failed, error:', err);
  });
}
```

**Comment by hungdt-ibl** [comment]
@8ctopotamus 
Thanks ! it worked 
but i still want to keep CRA worker for cached something

**Comment by hungdt-ibl** [comment]
i also get 
```
Module build failed (from ./node_modules/babel-loader/lib/index.js):
Error: Requires Babel "^7.0.0-0", but was loaded with "6.26.0". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel.
    at throwVersionError (/Volumes/Data/TECH/chatme/node_modules/@babel/helper-plugin-utils/lib/index.js:65:11)
    at Object.assertVersion (/Volumes/Data/TECH/chatme/node_modules/@babel/helper-plugin-utils/lib/index.js:13:11)
    at _default (/Volumes/Data/TECH/chatme/node_modules/@babel/plugin-transform-runtime/lib/index.js:82:7)
    at /Volumes/Data/TECH/chatme/node_modules/@babel/helper-plugin-utils/lib/index.js:19:12
    at Function.memoisePluginContainer (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/options/option-manager.js:113:13)
    at Function.normalisePlugin (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/options/option-manager.js:146:32)
    at /Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/options/option-manager.js:184:30
    at Array.map (<anonymous>)
    at Function.normalisePlugins (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at OptionManager.init (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/file/index.js:135:24)
    at Pipeline.transform (/Volumes/Data/TECH/chatme/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (/Volumes/Data/TECH/chatme/node_modules/babel-loader/lib/index.js:50:20)
    at Object.module.exports (/Volumes/Data/TECH/chatme/node_modules/babel-loader/lib/index.js:175:20)
 @ multi ./public/firebase-messaging-sw.js main[0]
    at compiler.run (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/index.js:93:11)
    at finalCallback (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/webpack/lib/Compiler.js:210:39)
    at hooks.done.callAsync.err (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/webpack/lib/Compiler.js:226:13)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/Hook.js:35:21)
    at onCompiled (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/webpack/lib/Compiler.js:224:21)
    at hooks.afterCompile.callAsync.err (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/webpack/lib/Compiler.js:553:14)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/Hook.js:35:21)
    at compilation.seal.err (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/webpack/lib/Compiler.js:550:30)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/Hook.js:35:21)
    at hooks.optimizeAssets.callAsync.err (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/webpack/lib/Compilation.js:1294:35)
    at AsyncSeriesHook.eval [as callAsync] (eval at create (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/HookCodeFactory.js:24:12), <anonymous>:6:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/tapable/lib/Hook.js:35:21)
    at hooks.optimizeChunkAssets.callAsync.err (/Volumes/Data/TECH/chatme/node_modules/cra-append-sw/node_modules/webpack/lib/Compilation.js:1285:32)
(node:6282) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:6282) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by tszarzynski** [comment]
@hungdt-ibl That's probably why the SW code is not being appended. I've seen this one before but I thought it was fixed already. Can you run the script with `--skip-compile` option? That should fix that error.

**Comment by renanccastro** [comment]
For those whose compile is failing, add the following dev dependency:

npm install -D babel-core@7.0.0-bridge.0

solved for me :+1:

**Comment by tszarzynski** [comment]
The dependencies issues should be fixed with #13.
